crates/biome_graphql_parser/src/lexer/mod.rs

```diff
@@ -0,0 +1,720 @@
+                    let diagnostic = ParseDiagnostic::new(
+                        "Invalid escape sequence",
+                        escape_start..self.text_position() + c.text_len(),
+                    )
+                    .with_hint(r#"For block string the only valid escape sequences is `\"""`. "#);
+                    (state, Some(diagnostic))
+                }
+            }
+            _ => panic!("String uninitialized/terminated"),
+        }
+    }
+
+    /// Lexes a `\u0000` escape sequence. Assumes that the lexer is positioned at the `u` token.
+    ///
+    /// A unicode escape sequence must consist of 4 hex characters.
+    fn consume_unicode_escape(&mut self) -> Result<(), ParseDiagnostic> {
+        self.assert_byte(b'u');
+        self.assert_current_char_boundary();
+
+        let start = self.text_position();
+
+        let start = start
+            // Subtract 1 to get position of `\`
+            .checked_sub(TextSize::from(1))
+            .unwrap_or(start);
+
+        self.advance(1); // Advance over `u`
+
+        for _ in 0..4 {
+            match self.current_byte() {
+                Some(byte) if byte.is_ascii_hexdigit() => self.advance(1),
+                Some(_) => {
+                    let char = self.current_char_unchecked();
+                    // Reached a non hex digit which is invalid
+                    return Err(ParseDiagnostic::new(
+
+                        "Invalid unicode sequence",
+                        start..self.text_position(),
+                    )
+                    .with_detail(self.text_position()..self.text_position().add(char.text_len()), "Non hexadecimal number")
+                    .with_hint("A unicode escape sequence must consist of 4 hexadecimal numbers: `\\uXXXX`, e.g. `\\u002F' for '/'."));
+                }
+                None => {
+                    // Reached the end of the file before processing 4 hex digits
+                    return Err(ParseDiagnostic::new(
+                        "Unicode escape sequence with two few hexadecimal numbers.",
+                        start..self.text_position(),
+                    )
+                    .with_detail(
+                        self.text_position()..self.text_position(),
+                        "reached the end of the file",
+                    )
+                    .with_hint("A unicode escape sequence must consist of 4 hexadecimal numbers: `\\uXXXX`, e.g. `\\u002F' for '/'."));
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn consume_comment(&mut self) -> GraphqlSyntaxKind {
+        assert_eq!(self.current_byte(), Some(b'#'));
+
+        self.advance(1);
+
+        while let Some(chr) = self.current_byte() {
+            match chr {
+                b'\n' | b'\r' => return COMMENT,
+                chr => self.advance_byte_or_char(chr),
+            }
+        }
+        COMMENT
+    }
+}
+
+fn is_name_start(byte: u8) -> bool {
```

Comment:
I'm wondering if we can use some functions from `crates/biome_unicode_table/src/lib.rs` here?

Reply:
I don't think so, those function support unicode characters while graphql identity only allow ascii